packages/react-aria-components/src/Calendar.tsx

```diff
@@ -466,7 +466,7 @@ function CalendarCell({date, ...otherProps}: CalendarCellProps, ref: ForwardedRe
 
   return (
     <td {...cellProps}>
-      <div {...mergeProps(buttonProps, focusProps, hoverProps, dataAttrs, renderProps)} ref={objectRef} />
+      <div {...mergeProps(filterDOMProps(otherProps as any), buttonProps, focusProps, hoverProps, dataAttrs, renderProps)} ref={objectRef} />
```

Comment:
The `as any` bothers me, but I couldn't come up with a good way to mark the props as only accepting data attributes since we've been relying on how TS doesn't consider a `data-*` attribute to be an error if it isn't found in the element attributes type. Also I've forgone the `delete DOMProps.id` when the props didn't share any values in common with the filterDOMProps type since we can reasonably expect only `data-*` attributes then.

Reply:
do cells not support an id at all? I guess I'm a little confused as to what the issue with id is in this whole situation

Reply:
yeah, we don't accept ids on the CalendarCell at all

Reply:
if they are only `data-*` and that is allowed on `CalendarCellProps`, you could create a type that would encompass it. maybe this would work?
```
type OnlyDataAttributes = { 
  [key: `data-${string}`]: unknown;
}
```

Reply:
Saving progress in `update_types_for_data_attributes` branch, the type changes are a bit extensive for this